geode-docs/developing/transactions/directed_example.html.md.erb

```diff
@@ -0,0 +1,234 @@
+
+An expected use case operates on two regions within a transaction.
+For performance purposes the
+<%=vars.product_name%> transaction implementation requires that region entries
+of partitioned regions be colocated.
+See [Custom-Partitioning and Colocating Data](../partitioned_regions/overview_custom_partitioning_and_data_colocation.html) for details on how to colocate region entries.
+
+
+## Transaction within an Application
+
+An application/client uses the `CacheTransactionManager` API.
+This most basic code fragment shows the structure of a transaction,
+with its `begin` to start the transaction, `commit` to end the transaction,
+and handling of exceptions that these methods may throw.
+
+``` pre
+CacheTransactionManager txManager =
+          cache.getCacheTransactionManager();
+
+try {
+    txManager.begin();
+    // ... do transactional, region operations
+    txManager.commit();
+} catch (CommitConflictException conflict) {
+    // ... do necessary work for a transaction that failed on commit
+}
+```
+
+More details of a transaction appear in this next application/client
+code fragment example.
+In this typical transaction,
+the put operations must be atomic and two regions are involved.
+
+In this transaction, a customer's purchase is recorded.
+The `cash` region contains each customer's cash balance
+available for making trades.
+The `trades` region records each customer's balance spent on trades.
+
+If there is a conflict upon commit of the transaction,
+an exception is thrown, and this example tries again.
+
+```
+// inputs needed for this transaction; shown as variables for simplicity
+final String customer = "Customer1";
+final Integer purchase = Integer.valueOf(1000);
+
+// region set up shown to promote understanding
+Cache cache = new CacheFactory().create();
+Pool pool = PoolManager.createFactory()
+           .addLocator("localhost", LOCATOR_PORT)
+           .create("pool-name");
+Region<String, Integer> cash =
+           cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+           .setPoolName(pool.getName())
+           .create("cash");
+Region<String, Integer> trades = 
+           cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+           .setPoolName(pool.getName())
+           .create("trades");
+
+// transaction code
+CacheTransactionManager txmgr = cache.getCacheTransactionManager();
+boolean commitConflict = false;
+do {
+  try {
+    txmgr.begin();
+
+    // Subtract out the cost of the trade for this customer's balance
+    Integer cashBalance = cash.get(customer);
+    Integer newBalance = 
+       Integer.valueOf((cashBalance != null ? cashBalance : 0) - purchase);
+    cash.put(customer, newBalance);
+
+    // Add in the cost of the trade for this customer
+    Integer tradeBalance = trades.get(customer);
+    newBalance = 
+       Integer.valueOf((tradeBalance != null ? tradeBalance : 0) + purchase);
+    trades.put(customer, newBalance);
```

Comment:
should handle TransactionException here. if is TransactionDataNotColocatedException, user should not retry the transaction. However, user can retry the same transaction if got TransactionDataRebalancedException or TransactionDataNodeHasDepartedException.